tests/unit/services/test_provider.py

```diff
@@ -84,41 +84,60 @@ def test_setup_proxy_environment(
 
 
 @pytest.mark.parametrize(
-    ("install_snap", "environment", "snaps"),
+    ("environment", "snaps"),
     [
-        (True, {}, [Snap(name="testcraft", channel="latest/stable", classic=True)]),
-        (
-            True,
+        pytest.param(
+            {},
+            [Snap(name="testcraft", channel="latest/stable", classic=True)],
+            id="install-from-store-default-channel",
+        ),
+        pytest.param(
             {"CRAFT_SNAP_CHANNEL": "something"},
             [Snap(name="testcraft", channel="something", classic=True)],
+            id="install-from-store-with-channel",
         ),
-        (
-            True,
-            {"SNAP_NAME": "testcraft", "SNAP": "/snap/testcraft/x1"},
-            [Snap(name="testcraft", channel=None, classic=True)],
+        pytest.param(
+            {
+                "SNAP_NAME": "testcraft",
+                "SNAP_INSTANCE_NAME": "testcraft_1",
```

Comment:
What about trying this one without SNAP_INSTANCE_NAME, for when the user doesn't have that feature enabled?